api/ssrApollo.js

```diff
@@ -19,6 +19,23 @@ import { MULTI_AUTH_ANON, MULTI_AUTH_LIST } from '@/lib/auth'
 
 export default async function getSSRApolloClient ({ req, res, me = null }) {
   const session = req && await getServerSession(req, res, getAuthOptions(req))
+
+  // If there's a session, check if the user is deleted
+  let meUser = me
+  if (session?.user) {
+    const user = await models.user.findUnique({
+      where: { id: parseInt(session.user.id) }, // Convert string to int
+      select: { deletedAt: true }
+    })
+
+    // If the user is deleted, don't pass the session to the context
+    if (user?.deletedAt) {
+      meUser = null
+    } else {
+      meUser = session.user
+    }
+  }
+
```

Comment:
Mhh, if we could revoke sessions (#1017), we wouldn't need this additional session check.

Will this run on every request? Can this be put into our auth code? I think ideally, `getServerSession` would already do all kinds of session checks and just not return a session if the user is deleted. Else I'm worried we will miss places where we need to check this.

Reply:
I can do that instead yes. Way cleaner.